gluoncv/model_zoo/alpha_pose/fast_pose.py

```diff
@@ -0,0 +1,262 @@
+    def __init__(self, preact, num_joints, deconv_dim=256, norm_layer=nn.BatchNorm, norm_kwargs=None, **kwargs):
+        super(AlphaPose, self).__init__(**kwargs)
+        self.preact = preact
+        self.num_joints = num_joints
+
+        self.shuffle1 = PixelShuffle(2)
+        if norm_kwargs is None:
+            norm_kwargs = {}
+        self.duc1 = DUC(1024, inplanes=512, upscale_factor=2, norm_layer=norm_layer, **norm_kwargs)
+        self.duc2 = DUC(512, inplanes=256, upscale_factor=2, norm_layer=norm_layer, **norm_kwargs)
+
+        self.conv_out = nn.Conv2D(
+            channels=num_joints,
+            in_channels=128,
+            kernel_size=3,
+            strides=1,
+            padding=1,
+            weight_initializer=initializer.Uniform(scale=math.sqrt(1 / (128 * 3 * 3))),
+            bias_initializer=initializer.Uniform(scale=math.sqrt(1 / (128 * 3 * 3)))
+        )
+
+    def hybrid_forward(self, F, x):
+        x = self.preact(x)
+        x = self.shuffle1(x)
+        x = self.duc1(x)
+        x = self.duc2(x)
+        x = self.conv_out(x)
+        return x
+
+
+def get_alphapose(name, dataset, num_joints, pretrained=False, pretrained_base=True, ctx=mx.cpu(),
+                  norm_layer=nn.BatchNorm, norm_kwargs=None,
+                  root=os.path.join('~', '.mxnet', 'models'), **kwargs):
+    r"""Utility function to return AlphaPose networks.
+
+    Parameters
+    ----------
+    name : str
+        Model name.
+    dataset : str
+        The name of dataset.
+    pretrained : bool or str
+        Boolean value controls whether to load the default pretrained weights for model.
+        String value represents the hashtag for a certain version of pretrained weights.
+    ctx : mxnet.Context
+        Context such as mx.cpu(), mx.gpu(0).
+    root : str
+        Model weights storing path.
+
+    Returns
+    -------
+    mxnet.gluon.HybridBlock
+        The AlphaPose network.
+
+    """
+    if norm_kwargs is None:
+        norm_kwargs = {}
+    preact = FastSEResNet(name, norm_layer=norm_layer, **norm_kwargs)
+    if not pretrained and pretrained_base:
+        from ..model_zoo import get_model
+        base_network = get_model(name, pretrained=True, root=root)
+        _try_load_parameters(self=base_network, model=base_network)
+    net = AlphaPose(preact, num_joints, **kwargs)
+    if pretrained:
+        from ..model_store import get_model_file
+        full_name = '_'.join(('alpha_pose', name, dataset))
+        net.load_parameters(get_model_file(full_name, tag=pretrained, root=root))
+    else:
+        import warnings
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            net.collect_params().initialize()
+    net.collect_params().reset_ctx(ctx)
+    return net
+
+def alpha_pose_resnet101_v1b_coco(**kwargs):
+    from ...data import COCOKeyPoints
+    keypoints = COCOKeyPoints.KEYPOINTS
+    num_gpus = kwargs.pop('num_gpus')
```

Comment:
this have to be changed otherwise will cause problem
```python
num_gpus = kwargs.pop('num_gpus', None)
if num_gpus is not None and int(num_gpus) > 1:
    norm_layer = mx.gluon.contrib.nn.SyncBatchNorm
    norm_kwargs = {'use_global_stats': False, 'num_devices': int(num_gpus)}
else:
    norm_layer = mx.gluon.nn.BatchNorm
    norm_kwargs = {'use_global_stats': False}

return get_alphapose(
    name='resnet101_v1b', dataset='coco',
    num_joints=len(keypoints), norm_layer=norm_layer,
    norm_kwargs=norm_kwargs, **kwargs)
```

Reply:
Thanks, I will fix this.